plugins/filter/filters.py

```diff
@@ -208,22 +208,46 @@ def listener_properties(self, listeners_dict, default_ssl_enabled,
 
             if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER':
                 final_dict['listener.name.' + listener_name + '.sasl.enabled.mechanisms'] = 'OAUTHBEARER'
+                final_dict['listener.name.' + listener_name + '.principal.builder.class'] =\
+                    'io.confluent.kafka.security.authenticator.OAuthKafkaPrincipalBuilder'
+
+            if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER' and not oauth_enabled:
```

Comment:
should we specify here and rbac_enabled to get more clarity later as i think OAUTHBEARER is true when either we have rbac_enabled or we have oauth_enabled

Reply:
we can only have OAUTHBEARER if we have rbac_enabled or oauth_enabled or both, since this condition has not oauth_enabled, it automatically goes to rbac_enabled

Reply:
your idea is also correct but this is just to keep the conditions as less as possible